### PR TITLE
Chain and asset agnostic refactor

### PR DESCRIPTION
**Pull request by ahdzib-maya**
- Changed `Chain` to less strict `string` type.
- Moved all chain definitions to each package's `utils.ts`.
- Duplicated all asset definitions to each package's `utils.ts`.
- Removed `chain.ts` and all of its functions.
- Moved `isAssetNativeRune` to `xchain-thorchain`.
- Moved all imports from `xchain-util` to `xchain-*` regarding asset and chains.

Implicit suggestions with the changes:
- Avoid importing both *Asset** and **Chain*, use `Asset*.chain` if imported.
- Leave chain names with shorthand style (e.g. `BTCChain` instead of `BitcoinChain`)
- Have *Asset** and **Chain* definitions in `utils.ts` instead of `const.ts`. (Made more sense for me)
- Standardize all assets as *Asset[Uppercase shorthand of asset]*
- Standardize `utils.ts` in all packages.

### CODE REVIEW

File: packages/xchain-util/src/asset.ts [hunk 1]
```diff
@@ -165,48 +164,42 @@ export const formatAssetAmount = ({
...
-
-/**
- * Base "chain" asset of Binance chain.
+ * Base "chain" asset on bitcoin cash main net.
  *
  * Based on definition in Thorchain `common`
  * @see https://gitlab.com/thorchain/thornode/-/blob/master/common/asset.go#L12-24
  */
-export const AssetBNB: Asset = { chain: Chain.Binance, symbol: 'BNB', ticker: 'BNB', synth: false }
+export const AssetBCH: Asset = { chain: 'BCH', symbol: 'BCH', ticker: 'BCH', synth: false }
```
Comment (veado) [inline packages/xchain-util/src/asset.ts:172]: Remove all `const Asset{XYZ}` from `asset.ts`. If you do need these in tests, define it there, but don`t export. _I've read your message about at Discord..._

**Review by veado — CHANGES_REQUESTED** [review]
Looks good, some minor changes:
-Keep constants in `const.ts` everywhere
- Use `{XYZ}Chain` instead of `{XYZAsset.chain}` everywhere

**Review by veado — CHANGES_REQUESTED** [review]
Two last things:
-  `AssetAtom.chain` -> `GaiaChain` everywhere
- Pls bump packages + update CHANGELOG's

   - `xchain-util`: Bump `minor` version (breaking change)
   - other packages: Bump `patch` version + update its deps to latest `xchain-util`
   
   @mikewyszinski If you have a chance, pls review.

**Review by veado — APPROVED** [review]
:+1:

**Review by veado — CHANGES_REQUESTED** [review]
Last change request (just realized):
- Use latest `"xchain-util@0.12.0"` in `xchain-client`
- Bump `xchain-client` to `0.13.5`
- Use `xchain-client@0.13.5` in other packages

**Review by mikewyszinski — APPROVED** [review]

**Review by veado — APPROVED** [review]

**Pull request merged**